daemon/daemon.go

```diff
@@ -530,6 +530,9 @@ func (daemon *Daemon) newContainer(name string, config *containertypes.Config, i
 
 // GetByName returns a container given a name.
 func (daemon *Daemon) GetByName(name string) (*container.Container, error) {
+	if len(name) == 0 {
+		return nil, fmt.Errorf("Could not find entity for empty container name")
```

Comment:
Can you use `errors.NewBadRequestError` here. This will ensure the proper HTTP handling.
Maybe improve the error message with `No container name or ID supplied`.